client/src/main/java/io/split/client/impressions/ImpressionsManagerImpl.java

```diff
@@ -96,17 +95,21 @@ private ImpressionsManagerImpl(CloseableHttpClient client,
                 : new ImpressionListener.NoopImpressionListener();
 
         _operationMode = config.operationMode();
-        _addPreviousTimeEnabled = shouldAddPreviousTime();
-        _counter = _addPreviousTimeEnabled ? new ImpressionCounter() : null;
-        _isOptimized = _counter != null && shouldBeOptimized();
-        if (_isOptimized) {
+        ProcessImpressionFactory processImpressionFactory =  new ProcessImpressionFactory();
+        if (_config.impressionsMode().equals(Mode.OPTIMIZED)){
```

Comment:
you are asking twice for the impression mode, we should chat this in private to improve it.

Reply:
I removed the Factory class. Now we are checking the mode in the ImpressionManager class.